packages/cli/src/runners/task-runner-process-restart-loop-detector.ts

```diff
@@ -0,0 +1,73 @@
+import { Time } from '@/constants';
+import { TaskRunnerRestartLoopError } from '@/runners/errors/task-runner-restart-loop-error';
+import type { TaskRunnerProcess } from '@/runners/task-runner-process';
+import { TypedEmitter } from '@/typed-emitter';
+
+const RESTART_LOOP_COUNT_LIMIT = 5;
+const RESTART_LOOP_TIME_LIMIT = 5 * Time.seconds.toMilliseconds;
+
+export type TaskRunnerProcessRestartLoopDetectorEventMap = {
```

Comment:
Do we need to export this?

Reply:
It's part of the public interface of the class, so I'd say yes

Reply:
If something is not yet needed outside a module, should it be part of its public interface? 

I lean towards minimizing the public interface where possible.

Reply:
That doesn't change the fact that it is part of its public interface, because it extends a type that requires that type. I removed the export tho since it's not used